tests/test_dedup.py

```
@@ -0,0 +1,492 @@
+        result = _compute_hashes(img)
+        assert len(result) == 4
+
+
+class TestComputeMd5:
+    """Test _compute_md5() file hashing."""
+
+    def test_correct_digest(self, tmp_path):
+        """Verify MD5 digest matches expected value for known content."""
+        f = tmp_path / "test.bin"
+        f.write_bytes(b"hello world")
+        expected = hashlib.md5(b"hello world").hexdigest()
```

Ruff (0.15.2)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)